controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -199,6 +216,27 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 				break
 			}
 		}
+
+		// check if a cephblockpool resource exists for the desired storageconsumer and storageprofile.
+		if r.cephBlockPool.Name == "" {
+			cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
+			listOptions := &client.MatchingLabels{
+				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+				controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
+			}
+			if err := r.list(cephBlockPoolList, client.InNamespace(r.OperatorNamespace), listOptions); err != nil {
+				return reconcile.Result{}, err
+			}
```

Comment:
A small suggestion, not a mandatory ask: Inline the label selector in the list operation.
```suggestion
			cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
			if err := r.list(
				cephBlockPoolList, 
				client.InNamespace(r.OperatorNamespace), 
				&client.MatchingLabels{
					controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
					controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
				}
			); err != nil {
				return reconcile.Result{}, err
			}
```